src/pmacFilterControl.cpp

```diff
@@ -493,8 +494,10 @@ void PMACFilterController::_process_state_changes() {
 */
 void PMACFilterController::_process_singleshot_state() {
     // Complete if singleshot run has stablised
-    if (this->state_ == ControlState::ACTIVE &&
-        this->last_received_frame_ - this->last_processed_frame_ > STABILITY_THRESHOLD
+    if (this->state_ == ControlState::ACTIVE && (
+            this->last_received_frame_ - this->last_processed_frame_ > STABILITY_THRESHOLD ||
+            this->current_attenuation_ == 0
+        )
     ) {
```

Comment:
Rather than setting it to one maybe just get rid of `STABILITY_THRESHOLD` and do
```suggestion
    if (this->state_ == ControlState::ACTIVE && (
            this->last_received_frame_ > this->last_processed_frame_ || this->current_attenuation_ == 0
        )
    ) {
```